src/Gui/Tree.cpp

```diff
@@ -1033,6 +1054,39 @@ void TreeWidget::onFinishEditing()
     }
 }
 
+// add childs of the selected tree object to selection
+void TreeWidget::onSelectAllChildren()
+{
+    // we only have this context menu entry if the selection is within one document
+    // so we can get the active document once
+    auto doc = Gui::Application::Instance->activeDocument()->getDocument();
+    auto docName = doc->getName();
+    // if the selected object is a document
+    if (this->contextItem && this->contextItem->type() == DocumentType) {
+        DocumentItem* docitem = static_cast<DocumentItem*>(this->contextItem);
+        //App::Document* doc = docitem->document()->getDocument();
+        std::vector<App::DocumentObject*> obj = doc->getObjects();
+        for (std::vector<App::DocumentObject*>::iterator it = obj.begin(); it != obj.end(); ++it)
+            Selection().addSelection(docName, (*it)->getNameInDocument());
+    }
+    // it can be an object
+    else {
+        QList<QTreeWidgetItem*> items = this->selectedItems();
+        for (QList<QTreeWidgetItem*>::iterator it = items.begin(); it != items.end(); ++it) {
+            if ((*it)->type() == ObjectType) {
+                DocumentObjectItem* objitem = static_cast<DocumentObjectItem*>(*it);
+                App::DocumentObject* obj = objitem->object()->getObject();
+                auto subObjectList = obj->getOutList();
+                // if there are childs ass them so the selection

```

Comment:
Several typos in the comment: childs, ass, so

Reply:
fixed